semantic/mesh/multipolymesh.cpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2021 Melown Technologies SE
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * *  Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * *  Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "../mesh.hpp"
+
+namespace semantic
+{
+namespace lod2
+{
+inline Material string2Material(const std::string& s)
+{
+    Material res;
+    if(boost::conversion::try_lexical_convert<Material>(s, res)){
+        return res;
+    }
+    return Material::default_;
+}
+
+geometry::Mesh mesh(const geometry::MultiPolyMesh<std::string>& mpmesh,
+                    const MeshConfig& config,
+                    const math::Point3& origin)
+{
+    (void) config;
+    std::vector<std::string> faceLabels;
+    geometry::Mesh m = mpmesh.triangulateFaces(&faceLabels);
+
+    for (auto& v : m.vertices)
+    {
+        v += origin;
+    }
+
+    for (std::size_t i = 0; i < m.faces.size(); i++)
+    {
+        m.faces[i].imageId = +string2Material(faceLabels[i]);
```

Comment:
Interesting idea to override the unary plus operator to get the underlying numerical representation. In the standard (C++23) this function will be called `std::to_underlying`. Can we rename it to something more descriptive? `toUnderlying`\`toInteger`\`asInteger`? Thanks.

Reply:
I think this should be discussed with the author of the operator, @vaclavblazek
https://github.com/melowntech/libsemantic/blob/388bc4a638aaca62ef577b44c485e59cb38014b3/semantic/mesh.hpp#L88

Reply:
I've found it somewhere as an idiom how to covert enum to int.

Reply:
I would vote for a direct cast or a function called as Matyas suggests. I assume there's a multitude of such hacks to get the underlying type from a typed enum, this one requires a foreign user to search for the implementation of the overloaded operator. If we do not want to change old code, I suggest to use the direct cast in the new parts.

Reply:
please use what's already there -- this is not contest in changing existing code
be pedand in your OWN code, pls